### PR TITLE
Improve contributing guidelines on PR/Commit Title/Messages

### PR DESCRIPTION
**Pull request by prologic**

### CODE REVIEW

**Comment by prologic** [comment]
I put this PR up in the hopes for further discussion as this came up in a discussion internally. Probably more eyes needed on it for general consensus. I'm just putting what was said and codifying it.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -183,13 +183,24 @@ One thing you will need to do only for your first pull request in Netdata is to
 
 #### PR title guidelines
 
-All verbs in PR titles must be in the past tense. E.g. "Added" instead of "Adds", "Fixed" instead of "Fixes".
+- All PR titles and Commit messages _must_ follow the [Imperative Mood](https://en.wikipedia.org/wiki/Imperative_mood)
+- Be no more than ~50 characters (_longer description in the PR_)
```
Comment (cosmix) [inline CONTRIBUTING.md]: Yes! I would stress the point that the PR description MUST contain sufficient information regarding the content of the PR, including area/component, tests done, etc.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -183,13 +183,24 @@ One thing you will need to do only for your first pull request in Netdata is to
...
-Improvements must have the title "[Area or module] : [Added/Refactored/Improved/Corrected/Started to/Stopped/Modified/Optimized] [x], [in order to/so that] [benefit]". The second part may be left out, if the benefit is obvious from the improvement "x".  
+- Fix bug in NetData installer for FreeBSD 11.2
+- Improve docs for other installation methods
+- New collector for Prometheus endpoints
+- Add 4.19 Kernel variant for eBPF
+- Fix typo in README
+- etc
 
-New collectors must have the title "[New collector] : [Name of collector]".
+For good examples have a look at other projects like:
```
Comment (cosmix) [inline CONTRIBUTING.md]: These are some of the most starred projects on GitHub, with hundreds of thousands of stars. I would also add:

the linux kernel itself and git among others. Note that even for commits, if the change is substantial, there should be a description (not in the title!) that explains the commit content. See this https://chris.beams.io/posts/git-commit/ — maybe worth mentioning it here.

Comment (prologic) [reply CONTRIBUTING.md]: i agree with you in general and I do this myself; but as we squash commits from PRs once merged this information gets lost _I think_ -- I _believe_ its preserved in the PR/head but IIRC Merged PR(s) get `git` committed with the PR title as the message and PR description as the rest?

File: CONTRIBUTING.md (hunk 1 above)
Comment (prologic) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -181,15 +181,34 @@ However, it's not always that simple. Our [PR approval process](#pr-approval-pro
...
 One thing you will need to do only for your first pull request in Netdata is to accept the CLA. Until you do, the automated check for the CLA acceptance will be showing as failed. 
 
-#### PR title guidelines
+#### PR Guidelines
 
-All verbs in PR titles must be in the past tense. E.g. "Added" instead of "Adds", "Fixed" instead of "Fixes".
+PR Titles:
 
-Bug fixes must have the title "Fixed [description of the problem as a user would perceive it], by [explanation of the correction]". 
+- Must follow the [Imperative Mood](https://en.wikipedia.org/wiki/Imperative_mood)
```
Comment (cakrit) [inline CONTRIBUTING.md]: I don't agree, we'll need to replace all verbs when writing release notes. Do suggest having the release notes in the imperative mood as well?

Comment (prologic) [reply CONTRIBUTING.md]: Change to their past tense?

@cosmix Do you have some comments here?

Comment (joelhans) [reply CONTRIBUTING.md]: I think an argument can be made for both past or present tense, so I won't get too involved in that. I don't think changing the tense for release notes will be _too_ much of an issue moving forward, as Patti and I are going to be updating the release notes in an ongoing basis as the sprints move along. It's a lot less daunting to do it piecemeal than in one giant batch like you did for v1.20, Chris. Just a thought.

**Comment by prologic** [comment]
@cosmix @cakrit Sorry for delay on this PR; Are there any other changes you want here?

**Pull request closed without merging**